GeneratorChain.cpp

```diff
@@ -0,0 +1,49 @@
+#include "GeneratorChain.h"
+
+// ================================================================================
+GeneratorChain::GeneratorChain()
+{
+    std::cout << "GeneratorChain constructor launched" << std::endl;
+    FormantAOUI.getFilteringAlgorithm(true);
+}
+
+GeneratorChain::~GeneratorChain()
+{
+    std::cout << "GeneratorChain destructor launched" << std::endl;
+    std::cout << "Last filter used: " << FormantAOUI.filterName << std::endl;
+}
+// ================================================================================
+
+float GeneratorChain::calculateTotalLevel(const OscillatorSection& osc, const FilterSection& filter) const
+{
+    std::cout << "Calculating total level..." << std::endl;
+    return osc.level * filter.gain;
+}
+
+void GeneratorChain::fadeInFilterMix(FilterSection& filter)
+{
+    if (filter.mix != 0.0f)
+    {
+        float temp = filter.mix;
+        filter.mix = 0.0f;
+        // for (int i = 0; i <= (int) (temp * 10); i++)
+        //     {
+        //         filter.mix += 0.1f;
+        //         std::cout << filter.mix << std::endl;
+        //     }
+        for (float i = 0.0f; i <= temp + 0.1f; i += 0.1f)    // not really a common usage of an iterator but it works :)
+        {
+            filter.mix = i;
+            std::cout << "Current filter mix amount: " << filter.mix << std::endl;            
+        }
+    }
+    else std::cout << "Error. Mix amount is 0.0f" << std::endl;
```

Comment:
If you use `if() { }` your `else` block should also use `{ }` even if the code in the curly braces is a single line of code
- [ ] revise